experimental/devices/cap1188/cap1188.go

```diff
@@ -0,0 +1,554 @@
+// Copyright 2017 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package cap1188 controls a Microchip cap1188 device over I²C.
+
+// The device is a 8 Channel Capacitive Touch Sensor with 8 LED Drivers
+//
+// Datasheet
+//
+// The official data sheet can be found here:
+//
+// http://ww1.microchip.com/downloads/en/DeviceDoc/CAP1188%20.pdf
+//
+package cap1188
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/mmr"
+	"periph.io/x/periph/conn/spi"
+	"periph.io/x/periph/devices"
+)
+
+var (
+	// Debug is a debugging flag to turn on extra logging
+	Debug         bool
+	inputStatuses []TouchStatus
+)
+
+// TouchStatus is the status of an input sensor
+type TouchStatus int8
+
+func (t TouchStatus) String() string {
+	switch t {
+	case OffStatus:
+		return "Off    "
+	case PressedStatus:
+		return "Pressed"
+	case HeldStatus:
+		return "Held   "
+	case ReleasedStatus:
+		return "Released"
+	default:
+		return "Unknown "
+	}
+}
+
+const (
+	// OffStatus indicates that the input sensor isn't being activated
+	OffStatus TouchStatus = iota
+	// PressedStatus indicates that the input sensor is currently pressed
+	PressedStatus
+	// HeldStatus indicates that the input sensor was pressed and is still held pressed
+	HeldStatus
+	// ReleasedStatus indicates that the input sensor was pressed and is being released
+	ReleasedStatus
+)
+
+const (
+	nbrOfLEDs = 8
+)
+
+const (
+	// reg_LEDLinking - The Sensor Input LED Linking register controls whether a
+	// capacitive touch sensor input is linked to an LED output. If the
+	// corresponding bit is set, then the appropriate LED output will change
+	// states defined by the LED Behavior controls in response to the capacitive
+	// touch sensor input.
+	reg_LEDLinking = 0x72
+	// reg_LEDOutputControl - The LED Output Control Register controls the output
+	// state of the LED pins that are not linked to sensor inputs.
+	reg_LEDOutputControl = 0x74
+)
+
+// Dev is a handle to a cap1188.
+type Dev struct {
+	d          conn.Conn
+	regWrapper mmr.Dev8
+	isSPI      bool
+	opts       *Opts
+	mu         sync.Mutex
+}
+
+func (d *Dev) String() string {
+	return fmt.Sprintf("cap1188{%s}", d.d)
+}
+
+// Halt is a noop for the cap1188.
```

Comment:
optional: Maybe turn off the LEDs?

Reply:
that would only work if the LEDs aren't linked. Maybe I can reset on halt? But that seems a bit odd

Reply:
It's not a big deal either way. In theory that would be a good place to put the device in deep sleep, just a TODO is good for now.